File: alphabet_spam.py
import sys

def ratios(string):
    length = len(string)
    whitespace_count = 0
    lower_count = 0
    upper_count = 0
    symbol_count = 0
    for char in string:
        if ord(char) == 95:
            whitespace_count += 1
        elif ord(char) >= 65 and ord(char) <= 90:
            upper_count += 1
        elif ord(char) >= 97 and ord(char) <= 122:
            lower_count += 1
        else:
            symbol_count += 1

    print(float(whitespace_count / length))
    print(float(lower_count / length))
    print(float(upper_count / length))
    print(float(symbol_count / length))



if __name__ == "__main__": 
    string_in = sys.stdin.readline()
    string_in.strip()
    ratios(string_in)

File: alphabet_spam.java
import java.util.Scanner;

public class alphabet_spam {

    public static void main(String[] args) {
        Scanner scnr = new Scanner(System.in);
        String input = scnr.nextLine();
        scnr.close();

        ratios(input);
    }

    public static void ratios(String string_in) {
        int whitespace_count = 0;
        int lower_count = 0;
        int upper_count = 0; 
        int symbol_count = 0; 
        int len = string_in.length();

        char[] str_arr = string_in.toCharArray();

        for (char letter : str_arr) {
            if (letter == 95) {
                whitespace_count++; 
            }
            else if (letter >= 97 && letter <= 122) {
                lower_count++; 
            }
            else if (letter >= 65 && letter <= 90) {
                upper_count++; 
            }
            else {
                symbol_count++; 
            }
        }

        System.out.println((double) whitespace_count / len);
        System.out.println((double) lower_count / len); 
        System.out.println((double) upper_count / len); 
        System.out.println((double) symbol_count / len); 
    }

}
